scopes/component/isolator/isolator.main.runtime.ts

```diff
@@ -336,12 +341,31 @@ export class IsolatorMain {
         .readdirSync(datedCapsuleDir, { withFileTypes: true })
         .filter((dir) => dir.isDirectory() && dir.name !== 'node_modules');
       allDirs.forEach((dir) => {
+        const sourceDir = path.join(datedCapsuleDir, dir.name);
+        const sourceCapsuleReadyFile = this.getCapsuleReadyFilePath(sourceDir);
+        if (!fs.pathExistsSync(sourceCapsuleReadyFile)) {
+          // Capsule is not ready, don't copy it to the cache
+          this.logger.console(`skipping moving capsule to cache as it is not ready ${sourceDir}`);
+          return;
+        }
         const targetDir = path.join(targetCapsuleDir, dir.name);
-        if (!fs.pathExistsSync(path.join(targetCapsuleDir, dir.name))) {
-          const sourceDir = path.join(datedCapsuleDir, dir.name);
-          this.logger.info(`moving specific capsule from ${sourceDir} to ${targetDir}`);
-          fs.moveSync(sourceDir, targetDir);
+        if (fs.pathExistsSync(path.join(targetCapsuleDir, dir.name))) {
+          const targetCapsuleReadyFile = this.getCapsuleReadyFilePath(targetDir);
+          if (fs.pathExistsSync(targetCapsuleReadyFile)) {
+            // Capsule is already in the cache, no need to move it
+            this.logger.console(`skipping moving capsule to cache as it is already exist at ${targetDir}`);
+            return;
+          }
+          this.logger.console(`cleaning target capsule location as it's not ready at: ${targetDir}`);
+          fs.removeSync(targetDir);
         }
+        this.logger.console(`moving specific capsule from ${sourceDir} to ${targetDir}`);
+        // We delete the ready file path first, as the move might take a long time, so we don't want to move
+        // the ready file indicator before the capsule is ready in the new location
+        this.removeCapsuleReadyFileSync(sourceDir);
```

Comment:
what happens if the process is killed in the middle and an orphan is ready file is left?